core/src/components/modal/modal.tsx

```diff
@@ -64,6 +65,8 @@ export class Modal implements ComponentInterface, OverlayInterface {
   private sortedBreakpoints?: number[];
   private keyboardOpenCallback?: () => void;
   private moveSheetToBreakpoint?: (options: MoveSheetToBreakpointOptions) => void;
+  // Whether or not the modal is moving to a new breakpoint
+  private breakpointMoving?: boolean;
```

Comment:
Would it make sense to leverage `currentTransition` for this instead? This would take some adjustments (having `moveSheetToBreakpoint` return a Promise, for example), but it would mean one less variable to worry about and allow other transition functions (such as `present` and `dismiss`) to avoid running into the breakpoint move animation without extra work. It may also let you get rid of `onAnimationFinish`, since I think you could resolve the promise in `moveSheetToBreakpoint` in the `onFinish` callback. Something like that anyway.